File: leetcode/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buy = prices[0]

        for price in prices[1:]:
            sell = price

            if (sell - buy < 0):
                buy = sell
            
            profit = max(profit, sell - buy)
        
        return profit

File: leetcode/best-time-to-buy-and-sell-stock.cpp
#include <algorithm>

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int buy = prices[0];

        for (int idx = 1; idx < prices.size(); idx++) {
            int sell = prices[idx];

            if (buy > sell) {
                buy = sell;
                continue;
            }

            profit = std::max(profit, sell - buy);
        }

        return profit;
    }
};